xapian-core/include/xapian/mset.h

```diff
@@ -74,6 +77,30 @@ class XAPIAN_VISIBILITY_DEFAULT MSet {
     /// Destructor.
     ~MSet();
 
+    /** Assigns new weights to items
+     * This functon is used in letor to update the weights assigned by the ranker
+     * It updates weights of all the elements in items as well as updating the max_attained
+     */
+    template <typename Iterator>
+    void replace_weights(Iterator first, Iterator last) {
+	if (std::distance(first, last) != size())
+	{
```

Comment:
Xapian coding standard is that `{` should be on the same line as `if`, etc.  Also `for` a few lines below.